File: baselibrary/src/main/java/com/appbaselib/common/ImageLoader.java
package com.appbaselib.common;

import android.content.Context;
import android.net.Uri;
import android.widget.ImageView;

import com.bumptech.glide.Glide;
import com.bumptech.glide.load.resource.drawable.GlideDrawable;
import com.bumptech.glide.request.RequestListener;

/**
 * Created by CLOUD on 2016/10/14.
 */

public class ImageLoader {

    public static void load(Context context, String path, ImageView imageView) {
        Glide.with(context).load(path)
                //  .centerCrop()  //
                .crossFade()
//                .placeholder(R.drawable.image_loading)
//                .error(R.drawable.image_failure)
                .into(imageView);

    }
    public static void load(Context context, Uri path, ImageView imageView) {
        Glide.with(context).load(path)
                //  .centerCrop()  //
                .crossFade()
//                .placeholder(R.drawable.image_loading)
//                .error(R.drawable.image_failure)
                .into(imageView);

    }

    public  static  void  load(Context context, String path, ImageView imageView, RequestListener<String,GlideDrawable> mRequestListener)
    {
        Glide.with(context).load(path)
                .crossFade()
                .listener(mRequestListener)
                .into(imageView);

    }

    public  static  void  load(Context context, Uri path, ImageView imageView, RequestListener<Uri,GlideDrawable> mRequestListener)
    {
        Glide.with(context).load(path)
                .crossFade()
                .listener(mRequestListener)
                .into(imageView);

    }
}

File: baselibrary/src/main/java/com/appbaselib/common/ImageLoader.kt
package com.appbaselib.common

import android.content.Context
import android.graphics.drawable.Drawable
import android.net.Uri
import android.widget.ImageView

import com.bumptech.glide.Glide
import com.bumptech.glide.load.resource.bitmap.CenterCrop
import com.bumptech.glide.load.resource.drawable.DrawableTransitionOptions.withCrossFade
import com.bumptech.glide.request.RequestListener
import com.bumptech.glide.request.RequestOptions
import com.pangu.appbaselibrary.R

/**
 * Created by CLOUD on 2016/10/14.
 */

object ImageLoader {
    fun load(context: Context, path: String, imageView: ImageView) {
        Glide.with(context).load(path)
                //  .centerCrop()  //
                //                .error(R.drawable.image_failure)
                .transition(withCrossFade())
                .into(imageView)

    }

    fun load(context: Context, path: String, imageView: ImageView,corner:Float) {
        Glide.with(context).load(path)
                //  .centerCrop()  //
                .apply(RequestOptions().transforms(CenterCrop(), RoundCorner(context, 12f)))
             //   .transition(withCrossFade())
                .into(imageView)

    }


    fun load(context: Context, path: Uri, imageView: ImageView) {
        Glide.with(context).load(path)
                //  .centerCrop()  //
                .transition(withCrossFade())
                //                .error(R.drawable.image_failure)
                .into(imageView)

    }

    fun load(context: Context, path: String, imageView: ImageView, mRequestListener: RequestListener<Drawable>) {
        Glide.with(context).load(path)
                .transition(withCrossFade())
                .listener(mRequestListener)
                .into(imageView)

    }

    fun load(context: Context, path: Uri, imageView: ImageView, mRequestListener: RequestListener<Drawable>) {
        Glide.with(context).load(path)
                .transition(withCrossFade())
                .listener(mRequestListener)
                .into(imageView)

    }
}
